crates/driver/src/domain/competition/order/mod.rs

```diff
@@ -133,6 +133,38 @@ impl Order {
         matches!(self.kind, Kind::Liquidity)
     }
 
+    /// The available sell amount for the order.
+    ///
+    /// This returns the maximum partial sell amount that can be used given past
+    /// partial executions and available balance.
+    pub fn available_sell(&self) -> eth::Asset {
```

Comment:
Wouldn't it be nicer to call this function in the auction code that computes max_sell (and also expose available_fee on the order)?

Reply:
We use it in #1861 (since we need to pass this to the solvers, and not just in the auction code that calls it). I prematurely put the function here with the knowledge that I would need it later - but I agree with your point that in a vacuum, it should go in the `auction` code that computes the max sell.

Reply:
We don't however use it in the `auction.rs`, there we still define `fn max_sell(order: &competition::Order)` inline. My point is that method should be replaced by calling `order.available_sell() + order.available_fee()` and avoid duplicating the `mul_ratio` logic.

Reply:
https://github.com/cowprotocol/services/pull/1861/files#r1339748474

This is what you are referring to right? I thought I had pushed this but if failed (because I did an amend commit 🤦).